### PR TITLE
Added functionality to abort text file processing when too many consecutive errors are encountered #2064

### PR DESCRIPTION
**Pull request by Onager**
Added functionality to abort text file processing when too many consecutive errors are encountered #2064

### CODE REVIEW

**Comment by Onager** [comment]
Reassigning to @joachimmetz as @rgayon is not available at present.

File: plaso/parsers/text_parser.py [hunk 1]
```diff
@@ -582,6 +594,10 @@ class PyparsingMultiLineTextParser(PyparsingSingleLineTextParser):
 
   BUFFER_SIZE = 2048
 
+  # The maximum number of consecutive lines that don't match known line
+  # structures to encounter before aborting parsing.
+  MAXIMUM_CONSECUTIVE_LINE_FAILURES = 20
```
Comment (rgayon) [inline plaso/parsers/text_parser.py]: Since PyparsingMultiLineTextParser inherits from PyparsingSingleLineTextParser, can you remove it here?

Comment (Onager) [reply plaso/parsers/text_parser.py]: That's a good point, I can indeed.

File: plaso/parsers/text_parser.py [hunk 2]
```diff
@@ -638,6 +634,8 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
     for key, structure in self.LINE_STRUCTURES:
       structure.parseWithTabs()
 
+
+    consecutive_line_failures = 0
```
Comment (rgayon) [inline plaso/parsers/text_parser.py]: ugh, didn't catch those... good call
Maybe test for this in a later PR ? =)

**Comment by joachimmetz** [comment]
@Onager please update this PR with recent changes to have tests pass

**Pull request merged**